### ISSUE TITLE
Failed installation in Ynh 4

### ISSUE THREAD

**Issue by Taekiro** [YunoHost-Apps/collabora_ynh#34]
Hello,
I've just migrated from Yunohost 3 to 4, and now I seems that I cannot install Collabora. Here is the log of one of the fail attempt : https://paste.yunohost.org/raw/pataguyoyu
Regards,
Taekiro.

**Comment by arthurlogilab** [comment]
Getting a very similar error.

**Comment by wget** [comment]
It seems like this file is missing `/etc/loolwsd/ca-chain.cert.pem` from the install process and the loolwsd service is therefore being restarted in an infinite loop which is killed after some time.

This file is, according to Yunohost scripts, populated via letsencrypt.

If you want to make it work, Collabora Online must be set to a *dedicated* subdomain.

This article on Nextcloud's Discourse instance may help you force the ssl certificate renewal: https://help.nextcloud.com/t/collabora-fails-to-start-systemctl-start-loolwsd/46070/3

@arthurlogilab Could you confirm whether following the steps described int he Nextcloud article does actually help?

**Comment by arthurlogilab** [comment]
@wget With some adaptation I'm getting it to work : 

```sh
openssl genrsa -out /etc/loolwsd/key.pem 4096
openssl req -batch -new -x509 -sha512 -days 3650 -key /etc/loolwsd/key.pem -out /etc/loolwsd/ca-chain.cert.pem
cp /etc/loolwsd/ca-chain.cert.pem /etc/loolwsd/cert.pem
chmod a+r  /etc/loolwsd/key.pem 
```

The last line is probably bad for security, it probably needs to be fixed with `chmod 0600 /etc/loolwsd/key.pem && chown lool /etc/loolwsd/key.pem` after the installation has created the `lool` user. 

By the way : when the ynh install fails, it deletes the /etc/loolwsd/ existing data (don't know if it's the install script or the debian package).

**Comment by wget** [comment]
@arthurlogilab Great! Thanks a lot for this technical feedback. This will help for future development to see how things behave in a ynh major update process.

Could you confirm to me under which domain Collabora Office and the Nextcloud installation have been defined from your side? I assume Nextcloud was on `example.org` and loolwsd on something like `loolwsd.example.org`, i.e. a dedicated subdomain from Nextcloud "main" domain, right?

**Comment by arthurlogilab** [comment]
The install succeed but I'm having trouble getting it to work in the browser and in nextcloud. Yes I have two separate domains for collabora and nextcloud. 

In the nextcloud the only configuration that works on the configuration side is pointing to `http://localhost:9980` but then when opening a document the iframe can't connect.

**Comment by alexAubin** [comment]
Ugh isn't there any way to tell Collabora to not handle at all any SSL-related stuff ? x_x

Or if it really needs to have some key/cert PEM file for some reason, would it be okay to create symlinks to /etc/yunohost/certs/domain.tld/ where Yunohost's key/cert files for that domain are stored ?

**Comment by alexAubin** [comment]
(Btw, pretty unrelated but since @wget you seems to be related to the dev team of Collabora somehow, dunno if the .deb package still includes a hard "systemctl restart nginx" but that'd be nice to only reload it instead of hard restart (or whatever mechanism to tweak that easily) because currently that breaks the install of Collabora from Yunohost's webadmin because the webadmin is ran through nginx and ugh)

**Comment by arthurlogilab** [comment]
Forgot to mention that I got it working (with a  nasty hairpinning / DNS fix in `/etc/hosts` where I had to add `collabora.example.org` pointing to `127.0.0.1` ... but I think that's a problem with my install on an internal network with dns problems).

**Comment by Taekiro** [comment]
Hello,
I just tried once more (in order to try @alexAubin solution with symlink to yunohost's cert), and I feel like the install changed (on Collabora's side ?). Here is the log, i don't have the "File not found: /etc/loolwsd/ca-chain.cert.pem" error anymore : https://paste.yunohost.org/raw/sapecesima .
Regards,
Taekiro.

**Comment by Taekiro** [comment]
Hi,
I don't know if it was supposed to be a fix, so here is my log for the testing branch updated last night  : https://paste.yunohost.org/raw/iguqicedop
Best regards,
Taekiro.

**Comment by Taekiro** [comment]
Hello,
I've done a new attempt to install the app since it changed a bit, and I think I have a new error : https://paste.yunohost.org/raw/axanopilop
Regards,
Taekiro.

**Comment by wget** [comment]
> Forgot to mention that I got it working (with a nasty hairpinning / DNS fix in `/etc/hosts` where I had to add `collabora.example.org` pointing to `127.0.0.1` ... but I think that's a problem with my install on an internal network with dns problems).

On the Docker articles I'm currently rewriting, I noticed a similar issue. From my understanding, when you link the CODE/Collabora Online instance to Nextcloud, you cannot use the domain name, because the latter is by default (nowadays) resolving to an IPv6 address hence the connectivity issue. However fixing the IP address manually seemed to work. May be related to a `gethostbyname` call somewhere. I unfortunately have more blocker tasks to fix this by myself, but I added this to my back log ;)
